src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/RuntimeTypeBuilder.cs

```diff
@@ -945,6 +946,10 @@ public override bool IsSubclassOf(Type c)
 
             while (p != null)
             {
+                // We can't use IsTypeEqual(p, this) because it would be recursive at least for Enums.
```

Comment:
I think it would be better to just return `false` instead of throwing ArgumentException.

ArgumentException would be fine to throw from SetParent, but that has its own set of problems (it would be a breaking change).